cupyx/scipy/special/_expn.py

```diff
@@ -0,0 +1,223 @@
+__constant__ double A12[] = {-479001600.000000000, 7827719040.00000000, -40788301824.0000000, 92199790224.0000000, -101180433024.000000, 56041398784.0000000, -15548960784.0000000, 2051482776.00000000, -114876376.000000000, 2170626.00000000000, -8166.00000000000000, 1.00000000000000000};
+__constant__ double *A[] = {A0, A1, A2, A3, A4, A5, A6, A7, A8, A9, A10, A11, A12};
+__constant__ double Adegs[] = {0, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11};
+
+/* Asymptotic expansion for large n, DLMF 8.20(ii) */
+__device__ double expn_large_n(int n, double x)
+{
+    int k;
+    double p = n;
+    double lambda = x/p;
+    double multiplier = 1/p/(lambda + 1)/(lambda + 1);
+    double fac = 1;
+    double res = 1; /* A[0] = 1 */
+    double expfac, term;
+
+    expfac = exp(-lambda*p)/(lambda + 1)/p;
+    if (expfac == 0) {
+        return 0;
+    }
+
+    /* Do the k = 1 term outside the loop since A[1] = 1 */
+    fac *= multiplier;
+    res += fac;
+
+    for (k = 2; k < nA; k++) {
+        fac *= multiplier;
+        term = fac*polevl<Adegs[k]>(lambda, A[k]);
+        res += term;
+        if (fabs(term) < MACHEP*fabs(res)) {
+            break;
+        }
+    }
+
+    return expfac*res;
+}
+
+'''
+
+
+expn_definition = (
+    polevl_definition
+    + gamma_definition
+    + expn_large_n_definition
+    + '''
+
+// include for CUDART_NAN, CUDART_INF
+#include <cupy/math_constants.h>
+
+__device__ double expn(int n, double x)
+{
+    double ans, r, t, yk, xk;
+    double pk, pkm1, pkm2, qk, qkm1, qkm2;
+    double psi, z;
+    int i, k;
+    double big = BIG;
+
+    if (isnan(x)) {
+        return CUDART_NAN;
+    } else if (n < 0 || x < 0) {
+        return CUDART_NAN;
+    }
+
+    if (x > MAXLOG) {
+        return (0.0);
+    }
+
+    if (x == 0.0) {
+        if (n < 2) {
+            return CUDART_INF;
+        } else {
+            return (1.0 / (n - 1.0));
+        }
+    }
+
+    if (n == 0) {
+        return (exp(-x) / x);
+    }
+
+    /* Asymptotic expansion for large n, DLMF 8.20(ii) */
+    if (n > 50) {
+        ans = expn_large_n(n, x);
+        return (ans);
+    }
+
+    /* Continued fraction, DLMF 8.19.17 */
+    if (x > 1.0) {
+        k = 1;
+        pkm2 = 1.0;
+        qkm2 = x;
+        pkm1 = 1.0;
+        qkm1 = x + n;
+        ans = pkm1 / qkm1;
+
+        do {
+	    k += 1;
+	    if (k & 1) {
+	        yk = 1.0;
+	        xk = n + (k - 1) / 2;
+	    } else {
+	        yk = x;
+	        xk = k / 2;
+	    }
+	    pk = pkm1 * yk + pkm2 * xk;
+	    qk = qkm1 * yk + qkm2 * xk;
+	    if (qk != 0) {
+	        r = pk / qk;
+	        t = fabs((ans - r) / r);
+	        ans = r;
+	    } else {
+	        t = 1.0;
+	    }
+	    pkm2 = pkm1;
+	    pkm1 = pk;
+	    qkm2 = qkm1;
+	    qkm1 = qk;
+	    if (fabs(pk) > big) {
+	        pkm2 /= big;
+	        pkm1 /= big;
+	        qkm2 /= big;
+	        qkm1 /= big;
+	    }
+        } while (t > MACHEP);
+
+        ans *= exp(-x);
```

Comment:
Seems that you have to return `ans` without going through this block.

Reply:
Ahh, yes! You are right. Have added it :)